action/protocol/rewarding/fund.go

```diff
@@ -85,8 +96,18 @@ func (p *Protocol) Deposit(
 	if _, err := p.state(ctx, sm, _fundKey, &f); err != nil {
 		return nil, err
 	}
-	f.totalBalance = big.NewInt(0).Add(f.totalBalance, amount)
-	f.unclaimedBalance = big.NewInt(0).Add(f.unclaimedBalance, amount)
+	f.totalBalance.Add(f.totalBalance, amount)
+	f.unclaimedBalance.Add(f.unclaimedBalance, amount)
+	if !isZero(sgdAmount) {
+		f.unclaimedBalance.Sub(f.unclaimedBalance, sgdAmount)
+		if f.unclaimedBalance.Sign() == -1 {
+			return nil, errors.New("no enough available balance")
+		}
+		// grant sgd amount to receiver
+		if err := p.grantToAccount(ctx, sm, receiver, sgdAmount); err != nil {
```

Comment:
as the IIP-15 doc described, amount of gas-fee will transfer to SGD Treasury and can  withdraw by sending a claim action. but here it is directly transfer to the receiver?

Reply:
so the total `amount` is added to Treasury, and `sgdAmount` is internally allocated to the receiver (not claimed yet), the receiver later send a claim action to get the SGD reward
say, before this:
```
total = 100
unclaimed = 100
```
with `amount = 10, sgdAmount =3`:
```
total = 110
unclaimed = 107 (indicating 3 are granted, not yet claimed)
```
after user claim the 3 tokens:
```
total = 107
unclaimed = 107
```